bin/yk-config

```diff
@@ -14,7 +14,7 @@ fi
 #
 # FIXME: Once we know what breaks us, allow the user to use the regular -O0,
 # -O1, -O2, etc. clang flags like normal.
-POSTLINK_PASSES_STR=verify
+POSTLINK_PASSES_STR="instcombine<max-iterations=1;no-use-loop-info;no-verify-fixpoint>"
```

Comment:
In the commit message, I think you should say where this comes from and why it fixes it.

Reply:
I don't think I know the answer to either of those. Well, it comes from `opt`, so I presume it's the default pipeline way of running `instcombine`. Why it fixes it? :man_shrugging:

Reply:
becuase `opt` has those arguments in the upstream pipeline, and one of them (the fixpoint one) suppresses the yklua build error.

Reply:
I assume the `no-` things turn parts of `instcombine` off and that's the fix?

Reply:
Ah, it might be the `no-verify-fixpoint` that fixed it, cause that sounds like the error I got.

Reply:
Is `no-verify-fixpoint` on its own enough I wonder?

Reply:
Fixed in [2aae465](https://github.com/ykjit/yk/pull/892/commits/2aae46516b3f3f5efcb416ef4e0aea7d8589aa5c). Personally, I don't get much out of this comment, but see if that works for you.

Reply:
> Is no-verify-fixpoint on its own enough I wonder?

Can we check this or, at least, raise an issue to investigate further? It might be useful to know what the minimal fix is.

Reply:
I can check this on Monday. Just leave the PR open until then.

Reply:
Thanks!

Reply:
I wondered what `no-verify-fixpoint` was, because it sounds potentially bad (e.g. what if it's checking for correctness). Actually, it's not really bad https://reviews.llvm.org/D154579:

>In order to prevent accidental optimization regressions in the future, this implements a default-enabled verify-fixpoint option, which will make sure that the fix point has indeed been reached after a single iteration. This means that tests where this is not the case need to be explicitly annotated. The actual optimization pipeline will disable this option, as failure to reach the fix point is expected to happen there (in rare cases, as described above).

So really it's an internal check that `InstCombine` does a good optimisation job, but that's too strict a check, which is why it's disabled in the real pipeline. [There is an interesting philosophical debate to be had here about whether this should have been on/off by default, but that's outside our control!]

My guess based on this is that we might only need `no-verify-fixpoint` for our case, but now that I understand what it is, I think "do whatever the default pipeline does" (which has a couple of other things) is the right thing to do.